lib/ocpp/v201/charge_point.cpp

```diff
@@ -889,17 +889,59 @@ void ChargePoint::update_aligned_data_interval() {
     }
 }
 
-bool ChargePoint::is_change_availability_possible(const ChangeAvailabilityRequest& req) {
-    if (req.evse.has_value() and this->evses.at(req.evse.value().id)->has_active_transaction()) {
-        return false;
-    } else {
+bool ChargePoint::any_transaction_active(const std::optional<EVSE>& evse) {
+    if (!evse.has_value()) {
         for (auto const& [evse_id, evse] : this->evses) {
             if (evse->has_active_transaction()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    if (this->evses.at(evse.value().id)->has_active_transaction()) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+bool operational_status_matches_connector_status(ConnectorStatusEnum connector_status,
+                                                 OperationalStatusEnum operational_status) {
+    return (operational_status == OperationalStatusEnum::Inoperative and
+            connector_status == ConnectorStatusEnum::Unavailable) or
+           (operational_status == OperationalStatusEnum::Operative and
+            (connector_status != ConnectorStatusEnum::Unavailable or connector_status != ConnectorStatusEnum::Faulted));
```

Comment:
This last statement is always true since it can't be Unavailable and Faulted at the same time.

Also, maybe easier to read using a switch for one of the enums?